src/lib.rs

```diff
@@ -449,8 +449,24 @@ impl RustSemsimian {
             .chain(object_terms.iter())
             .cloned()
             .collect();
-        let all_terms_vec: Vec<String> = all_terms.into_iter().collect();
-        let term_label_map = get_labels(&db_path_str, &all_terms_vec).unwrap();
+
+        let mut ancestors_set: HashSet<String> = HashSet::new();
+
+        // Expand each term using closure
+        for term in &all_terms {
+            let ancestors = expand_term_using_closure(term, &self.closure_map, &self.predicates)
+                .into_iter()
+                .collect::<HashSet<String>>();
+            ancestors_set.extend(ancestors);
+        }
+
+        // Combine all_terms_vec with ancestors_set by first adding all terms to the HashSet
+        ancestors_set.extend(all_terms);
+
+        // Convert the HashSet back to a Vec to get a list of unique elements
+        let combined_unique_vec: Vec<String> = ancestors_set.into_iter().collect();
+
+        let mut term_label_map = get_labels(&db_path_str, &combined_unique_vec).unwrap();
```

Comment:
This code gets the ancestors and their corresponding labels avoiding the multiple db querying!